TCutility/analysis/ts_vibration.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+from TCutility.results import read
+from scm import plams
+
+def atom_distance(molecule: plams.Molecule, atom1: int, atom2: int) -> float:
+    res = 0.0
+    for i,j in zip(molecule[atom1],molecule[atom2]):
+        res += (i - j)**2
+    return np.sqrt(res)
+
+def avg_relative_bond_length_delta(base: plams.Molecule, pos: plams.Molecule,neg: plams.Molecule, a1: int, a2: int) -> float:
+    basedist = atom_distance(base, a1+1, a2+1)
+    x = abs((atom_distance(pos, a1+1, a2+1)/basedist)-1)
+    y = abs((atom_distance(neg, a1+1, a2+1)/basedist)-1)
+    return (x+y)/2
+
+def determine_ts_reactioncoordinate(calc_dir: str, bond_tolerance: float = 1.28, min_delta_dist: float = 0.1) -> np.ndarray:
+    '''Function to retrieve reaction coordinate from a given transitionstate, using the first imaginary frequency.
+
+    Args:
+        calc_dir: path pointing to the desired calculation.
+        bond_tolerance: parameter for plams.molecule.guess_bonds() function
+        min_delta_dist: minimum relative bond length change before qualifying as active atom
+
+    Returns:
+        Array containing all the obtained reaction coordinates. Reaction coordinate format is [active_atom1, active_atom2, sign]
+        Symmetry elements are ignored, by convention the atom labels are increasing (atom1 < atom2)
+    '''
+
+    data = read(calc_dir)
+    assert 'modes' in data.properties.vibrations, 'Vibrational data is required, but was not present in .rkf file'
```

Comment:
Also check if the frequencies are imaginary. Or you could loop through all modes that have imaginary frequencies and return a list of lists

Reply:
Add an argument to select a normal mode to analyse (default to 0). Also add a new function that analyses all imaginary vibrations for a given molecule